### PR TITLE
Test generic entities

### PR DESCRIPTION
**Pull request by FrankBakkerNl**
## Proposed change

A new throw at type safe entities and states

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/App/NetDaemon.App/Common/Reactive/Services/RxEntityBase.cs [hunk 1]
```diff
@@ -2,9 +2,28 @@
 using System.Collections.Generic;
 using System.Dynamic;
 using System.Linq;
+using System.Reactive.Linq;
+using NetDaemon.Common;
+using NetDaemon.Common.Reactive;
 
 namespace NetDaemon.Common.Reactive.Services
 {
+    public abstract class RxEntityBase<TEntityState> : RxEntityBase where TEntityState : IEntityProperties
```
Comment (FrankBakkerNl) [inline src/App/NetDaemon.App/Common/Reactive/Services/RxEntityBase.cs:11]: This inheritance chain is getting pretty long this way.

RxEntityBase :
RxEntityBase
RxEntity
IRxEntityBase

The upside is that this way we do not break existing code. Downside is that it becomes less and less clear which is responsible for what.

An alternative is to create a brand new EntityBase<> class besides the exisiting structure, have not looked into how hard that would be

Comment (helto4real) [reply src/App/NetDaemon.App/Common/Reactive/Services/RxEntityBase.cs:11]: Yea agree. We have to think about how to make a smooth transition . We could make a transitional deprecated version ... that inherits the new API.  I rather replace the old structure if we could and the publicly exposed interfaces deprecates for a couple of months. But yea you made a clear point. This needs refactoring!!

**Review by helto4real — COMMENTED** [review]
I like where this is heading. we could discuss and decide new design in Discord dev group.

**Comment by helto4real** [comment]
@FrankBakkerNl the CI seem to pass now when I restarted it. If @BeeHiveJava reviews it and clear we could merge this.

**Review by BeeHiveJava — APPROVED** [review]
Hey this looks good. Like you said @FrankBakkerNl, the inheritance chain is getting pretty nested. I think that's fine especially in this experimenting stage, it allows to keep the API backwards compatible. I'd like to get rid of the dynamics in `ClimateEntity` as well (those were already there).

I have some ideas on how to refactor entities & properties, I'll share them soon with some examples.

**Comment by helto4real** [comment]
Can we make this PR a draft @FrankBakkerNl ? So I know when you think it is ready to be merged.

**Comment by FrankBakkerNl** [comment]
Closing this, is superseded by other PR

**Pull request closed without merging**